1-4 задачи решать с помощью функций, 5ю вручную  
*Выберете тест и проверьте, есть ли различия между выборками:  

##### 1 ) Даны две независимые выборки. Не соблюдается условие нормальности  

    x1 380,420, 290
    y1 140,360,200,900
    
*Сделайте вывод по результатам, полученным с помощью функции*  

Две независимые выборки с ненормальным распределением - это тест Манна-Уитни, применим функцию в питоне:

In [5]:
import numpy as np
import scipy.stats as stats
x1=np.array([380,420, 290])
y1=np.array([140,360,200,900])
stats.mannwhitneyu(x1,y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

p-value больше 5%, значит, **верна нулевая гипотеза** о статитистической неразличимости этих выборок.

#### Задача 2 
Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?  
1е измерение до приема препарата: 150, 160, 165, 145, 155  
2е измерение через 10 минут: 140, 155, 150, 130, 135  
3е измерение через 30 минут: 130, 130, 120, 130, 125  

Так как давления мерялось на одних и тех же пациентах, то выборки считаем зависимыми.
Про нормальность и дисперсию ничего не сказано, поэтому можем сразу взять непараметрический тест Фридмана для нескольких выборок 

In [6]:
before_gr = [150,160,165,145,155]
min10_gr = [140, 155, 150, 130, 135]
min30_gr = [130, 130, 120, 130, 125]
stats.friedmanchisquare(before_gr,min10_gr,min30_gr)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

p-value равен 0.008, что значительно меньше статистической значимости в 0.05 , значит статистически значимые различия до и после препарата имеются.  

Ради интереса, посмотрим, как быстро это влияние себя показало для теста по двум выборкам(собственно, **Задача (3)**):

In [15]:
stats.wilcoxon(before_gr,min10_gr)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

In [19]:
stats.wilcoxon(before_gr,min30_gr)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

По гипотезе что выборки равны, pvalue  в 6,25% , что довольно мало, но все еще в пользу "нет отличий" при $\alpha$ в 5% . Также statistic получился равен 0 , что говорит о том, что все различия(дельты) у нас односторонние. Проверим по Уилкоксону альтернативную гипотезу:

In [16]:
stats.wilcoxon(before_gr,min30_gr,alternative="less")

WilcoxonResult(statistic=15.0, pvalue=1.0)

Со 100% вероятностью верна гипотеза, что медиана второй выборки меньше медианы начальной выборки.

In [20]:
stats.wilcoxon(min10_gr,min30_gr,zero_method='wilcox')

C:\Users\Evercrow\AppData\Roaming\Python\Python310\site-packages\scipy\stats\_morestats.py:3255: UserWarning: Exact p-value calculation does not work if there are zeros. Switching to normal approximation.
  warnings.warn("Exact p-value calculation does not work if there are "
C:\Users\Evercrow\AppData\Roaming\Python\Python310\site-packages\scipy\stats\_morestats.py:3269: UserWarning: Sample size too small for normal approximation.
  warnings.warn("Sample size too small for normal approximation.")


WilcoxonResult(statistic=0.0, pvalue=0.06559969214707187)

При сравнении двух выборок после введения  препарата получаем ошибку о слишком малом объеме выборок для правильного теста и наличии 0-й в дельтах.  
Вывод: тест Уилкоксона в этой задаче не слишком подходит, и не дает каких-то уверенных результатов, но можно говорить о большей похожести выборок по 10 и 30 минутам, чем о выборках до начала препарата и любой после.

#### Задача 4
Даны 3 группы учеников плавания.  
В 1 группе время на дистанцию 50 м составляют: 56, 60, 62, 55, 71, 67, 59, 58, 64, 67  
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53  
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54  

Группы у нас по разным ученикам, так что берем тест для нескольких независимых выборок, по Краскелу-Уоллису

In [21]:
st_grA = [56, 60, 62, 55, 71, 67, 59, 58, 64, 67]
st_grB = [57, 58, 69, 48, 72, 70, 68, 71, 50, 53]
st_grC = [57, 67, 49, 48, 47, 55, 66, 51, 54]
stats.kruskal(st_grA,st_grB,st_grC)

KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

Получаем, что с 6.5% вероятностью генеральная совокупность наших выборок будет распределяться по $\Chi^2$, что выше стандартной  $\alpha$ в 5% , а значит, различия между нашими группами студентов будут небольшие(и выборки взяты из одной генеральной совокупности)

#### Задача 5
Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%  
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

Дано:
* условие нормальности выполнено
* $\mu_0=2,5$ см
* $\alpha =5\%$
* $n=10$
* альтернативная гипотеза $H_1$ что $\mu \neq \mu_0$ 

Дисперсия генеральной совокупности нам неизвестна, но даны значения выборки, значит , будем работать по криттерию Стьюдента

In [22]:
parts=[2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]
np.std(parts)

0.14918444959177218

In [25]:
np.mean(parts)

2.5279999999999996

In [24]:
stats.t.ppf(1-0.05/2,len(parts)-1)

2.2621571627409915

Получаем среднее квадратичное отклонение по выборке в 0.15 ,среднее арифметическое 2.53,  и критическое значение по стьюденту с двусторонним тестом равноe 2.26

Теперь подставим это в формулу расчета криттерия Стьюдента:
$$ t_н = \frac{\bar{X}-\mu_0}{\sigma/\sqrt{n}}$$
$$ t_н = \frac{2.5-2.53}{0.15/\sqrt{10}}=\frac{0.03}{0.15/3.16}=0.632$$

Сравниваем $t_н$ с критическим значением $t_{\frac{\alpha}{2}}=2.26$ - он меньше, значит верна нулевая гипотеза о том, что истинной среднее арифметическое действительно равно 2.5 см

Проверим через функцию в питоне:

In [34]:
stats.ttest_1samp(parts,2.5)

Ttest_1sampResult(statistic=0.5630613661802959, pvalue=0.5871439993940629)

Тоже получаем большой p-value для нулевой гипотезе о верности среднего арифметического в 2.5 см